Lib/typing.py

```diff
@@ -2252,10 +2252,18 @@ def __new__(cls, name, bases, ns, total=True):
         Subclasses and instances of TypedDict return actual dictionaries.
         """
         for base in bases:
-            if type(base) is not _TypedDictMeta:
+            if not (type(base) is _TypedDictMeta or base is Generic):
                 raise TypeError('cannot inherit from both a TypedDict type '
                                 'and a non-TypedDict base class')
-        tp_dict = type.__new__(_TypedDictMeta, name, (dict,), ns)
+        
+        generic_base = tuple(b for b in bases if b is Generic)
+        base_typevars = _collect_type_vars(ns.get("__orig_bases__", ()))
+        if not generic_base and base_typevars:
+            # If not directly inheriting from a Generic 
+            # but inheriting from a generic TypedDict
+            generic_alias = Generic.__class_getitem__(base_typevars),
+            generic_base = types.resolve_bases(generic_alias)
+        tp_dict = type.__new__(_TypedDictMeta, name, (*generic_base, dict,), ns)
```

Comment:
I think this code can be simplified.

We should check that one of bases is subclass of `Generic` or is a `Generic` and in such case we should add `Generic` to list of base classes.

 ```suggestion
        generic_base = ()
        if any(issubclass(b, Generic) for b in bases):
            generic_base = (Generic,)

        tp_dict = type.__new__(_TypedDictMeta, name, (*generic_base, dict,), ns)
```

Reply:
Oh is it effectively the same? I'll give it a shot.

I'm working on adding the test cases.

Thank you for taking a look.

Reply:
I see what I was trying to do with collecting typevars from `orig_bases` is already handled nicely in `Generic.__init_subclass__`